### PR TITLE
Merge `8.2.x` into `develop`

### PR DESCRIPTION
**Pull request by tleon**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Merge `8.2.x` into `develop`
| Type?             | improvement 
| Category?         | ME 
| BC breaks?        | no
| Deprecations?     |  no
| How to test?      | CI & tests green
| UI Tests          | https://github.com/tleon/ga.tests.ui.pr/actions/runs/10925418491
| Sponsor company   | PrestaShop SA

### CODE REVIEW

File: classes/order/Order.php
Comment (Hlavtox) [inline classes/order/Order.php]:
```suggestion
        if (Configuration::get('PS_ADVANCED_STOCK_MANAGEMENT')
```

PS_ADVANCED_STOCK_MANAGEMENT Has been completely removed, you can keep only the else part

Comment (jolelievre) [reply classes/order/Order.php]: Indeed

Comment (Hlavtox) [reply classes/order/Order.php]: @tleon This was the change in the method. ;-)

https://github.com/PrestaShop/PrestaShop/commit/74d5c5c924350feab0e6b3707bdb1e9488252ccf

Just cast location to string.

**Review by Hlavtox — CHANGES_REQUESTED** [review]
Just to prevent accidental merge, dismiss when the feedback is implemented

**Review by boherm — DISMISSED** [review]

**Review by Hlavtox — DISMISSED** [review]

File: src/PrestaShopBundle/DependencyInjection/Config/ConfigYamlLoader.php
Comment (Hlavtox) [inline src/PrestaShopBundle/DependencyInjection/Config/ConfigYamlLoader.php]: Something weird here?

**Review by Hlavtox — DISMISSED** [review]
See comment

File: classes/Product.php
Comment (kpodemski) [inline classes/Product.php]: Looks like a mistake with the placement of `}`

File: classes/controller/AdminController.php
Comment (kpodemski) [inline classes/controller/AdminController.php]: Mistake?

Comment (tleon) [reply classes/controller/AdminController.php]: There is a lot of fail jobs i'm testing thing to see what make the tests go red.

**Review by boherm — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Review by Hlavtox — APPROVED** [review]

**Comment by ps-jarvis** [comment]
PR merged, well done!

Message to @PrestaShop/committers: do not forget to milestone it before the merge.

**Pull request merged**
